application/src/main/kotlin/net/corda/v5/application/crypto/DefaultSignatureSpecService.kt

```diff
@@ -0,0 +1,26 @@
+package net.corda.v5.application.crypto
+
+import net.corda.v5.crypto.SignatureSpec
+import java.security.PublicKey
+
+interface DefaultSignatureSpecService {
+    /**
+     * Work out a signature and hash algorithm to use for a given public key, given current security policies.
+     *
+     * @param publicKey the public key to be used for signing
+     *
+     * @return An appropriate [SignatureSpec], or null if nothing is available for the key type.
+     */
+    fun suggest(publicKey: PublicKey): SignatureSpec?
+
+    /**
+     * Work out a signature algorithm given current security policies and a hash algorithm
+     *
+     * @param publicKey the public key to be used for signing
+     * @param hash the hash algortithm e.g. "SHA256" - TODO: use an interface or enum here?
+     *
+     * @return An appropriate [SignatureSpec], or null if nothing is available for the key type
+     */
+
+    fun suggest(publicKey: PublicKey, hash: String): SignatureSpec?
+}
```

Comment:
We also need to add a `fun compatible(publicKey: PublicKey, hash: DigestAlgorithmName): List<SignatureSpec>` for either people who don't want to use our defaults, or for situations where older data used the default and we later changed that default

Reply:
That could potentially be done in a later step. I'm also not sure if the output should be a `Set<SignatureSpec>`, of if keep it as a list is the order significant, e.g. most secure first? @kyriathar any thoughts?

Reply:
`listCompatible` might be a better name?

Reply:
I would think that `fun compatible` should only take `PublicKey` because with `PublicKey` and `DigestAlgorithm` that can lead to a single `SignatureSpec` ?